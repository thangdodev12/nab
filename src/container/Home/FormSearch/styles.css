.search {
  position: relative;
}
.autocomplete {
  position: absolute;
  top: 0.5em;
  left: 0;
  background-color: #fff;
  box-shadow: 4px 4px 12px #ccc;
  padding: 0;
  list-style: none;
}

.autocomplete li {
  padding: 0.5em 2em;
}