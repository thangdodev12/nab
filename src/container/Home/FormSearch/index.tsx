import { useCallback } from 'react';
import { connect } from 'react-redux';
import { getLocationWeather as getLocationWeatherAction } from '../actions';
import './styles.css';

const FormSearch = ({ loading, getLocationWeather }: any) => {
  const onSubmit = useCallback((e) => {
    e.preventDefault();
    getLocationWeather(e.target.search.value);
  }, [getLocationWeather]);
  return (
    <form className="search" onSubmit={onSubmit}>
      <input type="search" className="search__input" name="search" placeholder="ex: Ho Chi Minh" />
      <button type="submit" className="search__button" disabled={loading}>Search</button>
    </form>
  );
}

const mapDispatchToProps = (dispatch: any) => ({
  getLocationWeather: (query: string) => dispatch(getLocationWeatherAction(query)),
});

export default connect(null, mapDispatchToProps)(FormSearch);
