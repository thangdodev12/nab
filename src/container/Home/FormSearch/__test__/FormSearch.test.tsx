import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import FormSearch from '../';

const mockStore = configureStore([]);

describe('FormSearch', () => {
  let props, store;
  beforeEach(() => {
    store = mockStore({
      home: {
        data: {
          title: 'Ho Chi Minh',
          consolidatedWeather: [],
          time: '2021-01-01'
        }
      },
    });
    props = {
      loading: false,
    }
  });

  it('should render match  snapshot', () => {
    const component = renderer.create(
      <Provider store={store}>
        <FormSearch {...props} />
      </Provider>,
    );
    const snapshot = component.toJSON();
    expect(snapshot).toMatchSnapshot();
  })
})
