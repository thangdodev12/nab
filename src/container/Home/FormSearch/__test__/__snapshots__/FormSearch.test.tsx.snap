// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSearch should render match  snapshot 1`] = `
<form
  className="search"
  onSubmit={[Function]}
>
  <input
    className="search__input"
    name="search"
    placeholder="ex: Ho Chi Minh"
    type="search"
  />
  <button
    className="search__button"
    disabled={false}
    type="submit"
  >
    Search
  </button>
</form>
`;

exports[`Home should render match  snapshot 1`] = `
<form
  className="search"
  onSubmit={[Function]}
>
  <input
    className="search__input"
    name="search"
    placeholder="ex: Ho Chi Minh"
    type="search"
  />
  <button
    className="search__button"
    disabled={false}
    type="submit"
  >
    Search
  </button>
</form>
`;
